vpdq/python/test_vpdq_hash.py

```diff
@@ -0,0 +1,50 @@
+import vpdq
+import typing as t
+import pytest
+import test_util
+import os
+import pathlib
+
+SAMPLE_HASH_FOLDER = "/vpdq/sample-hashes"
+SAMPLE_VIDEOS = "/tmk/sample-videos"
+FFMPEG = "/usr/bin/ffmpeg"  # please customize according to your installation (but do not commit)
+SECOND_PER_HASH = 1
+DISTANCE_TOLERANCE = 10
+QUALITY_TOLERANCE = 50
+
+
+class TestVPDQHash:
+    def test_VPDQ_hash(self):
+        d = os.path.dirname(os.path.dirname(os.getcwd()))
+        hash_folder = pathlib.Path(d + SAMPLE_HASH_FOLDER)
+        video_folder = pathlib.Path(d + SAMPLE_VIDEOS)
+        self.test_hashes = {}
+        self.sample_hashes = {}
+
+        for file in sorted(os.listdir(hash_folder)):
+            if file.endswith(".txt"):
+                hash_file = pathlib.Path(f"{hash_folder}/{file}")
+                ret = test_util.read_file_to_hash(hash_file)
+                name = os.path.splitext(file)[0]
+                self.sample_hashes[name] = ret
+
+        for file in sorted(os.listdir(video_folder)):
+            if file.endswith(".mp4"):
+                video_file = pathlib.Path(f"{video_folder}/{file}")
+                ret = vpdq.computeHash(
+                    input_video_filename=str(video_file),
+                    ffmpeg_path=FFMPEG,
+                    seconds_per_hash=SECOND_PER_HASH,
+                )
+                name = os.path.splitext(file)[0]
+                self.test_hashes[name] = ret
+
+        for k in self.sample_hashes.keys():
+            print("Comparing hash for video: " + k)
+            hash1 = self.test_hashes[k]
+            hash2 = self.sample_hashes[k]
+            assert len(hash1) == len(hash2)
+            for h1, h2 in zip(hash1, hash2):
```

Comment:
blocking: This is a bit too loose for me. We are relying on the sorted(list) of these files to exactly line up without checking if they actually line up.

Instead, can we establish a convention that we always expect a corresponding named file from the sample hashes, and assert it is there as part of the test?

I.e. instead of two for loops above, only have one for loop that detects one of the files, and then looks up the other by name. We then end up with a dict of name => (sample, test)

Reply:
The two loops are checking the hashes inside each VPDQ file, look up stage is actually:
```
for k in self.sample_hashes.keys():
            print("Comparing hash for video: " + k)
            hash1 = self.test_hashes[k]
            hash2 = self.sample_hashes[k]
            
```
I will add test to assert them not be none.